instrumentation/vertx/vertx-sql-client/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/QueryExecutorInstrumentation.java

```diff
@@ -54,71 +54,94 @@ public static void onExit(@Advice.This Object queryExecutor) {
+
+    public static class AdviceScope {
+      private final CallDepth callDepth;
+      @Nullable private final VertxSqlClientRequest otelRequest;
+      @Nullable private final Context context;
+      @Nullable private final Scope scope;
+
+      private AdviceScope(CallDepth callDepth) {
+        this(callDepth, null, null, null);
+      }
+
+      private AdviceScope(
+          CallDepth callDepth, VertxSqlClientRequest otelRequest, Context context, Scope scope) {
+        this.callDepth = callDepth;
+        this.otelRequest = otelRequest;
+        this.context = context;
+        this.scope = scope;
       }
 
-      // The parameter we need are in different positions, we are not going to have separate
-      // advices for all of them. The method gets the statement either as String or
-      // PreparedStatement, use the first argument that is either of these. PromiseInternal is
-      // always at the end of the argument list.
-      String sql = null;
-      PromiseInternal<?> promiseInternal = null;
-      for (Object argument : arguments) {
-        if (sql == null) {
-          if (argument instanceof String) {
-            sql = (String) argument;
-          } else if (argument instanceof PreparedStatement) {
-            sql = ((PreparedStatement) argument).sql();
+      public static AdviceScope start(Object queryExecutor, Object[] arguments) {
+        CallDepth callDepth = CallDepth.forClass(queryExecutor.getClass());
+        if (callDepth.getAndIncrement() > 0) {
+          return new AdviceScope(callDepth);
+        }
+
+        // The parameter we need are in different positions, we are not going to have separate
+        // advices for all of them. The method gets the statement either as String or
+        // PreparedStatement, use the first argument that is either of these. PromiseInternal is
+        // always at the end of the argument list.
+        String sql = null;
+        PromiseInternal<?> promiseInternal = null;
+        for (Object argument : arguments) {
+          if (sql == null) {
+            if (argument instanceof String) {
+              sql = (String) argument;
+            } else if (argument instanceof PreparedStatement) {
+              sql = ((PreparedStatement) argument).sql();
+            }
+          } else if (argument instanceof PromiseInternal) {
+            promiseInternal = (PromiseInternal<?>) argument;
           }
-        } else if (argument instanceof PromiseInternal) {
-          promiseInternal = (PromiseInternal) argument;
         }
+        if (sql == null || promiseInternal == null) {
+          return new AdviceScope(callDepth);
+        }
+
+        VertxSqlClientRequest otelRequest =
+            new VertxSqlClientRequest(sql, QueryExecutorUtil.getConnectOptions(queryExecutor));
+        Context parentContext = Context.current();
+        if (!instrumenter().shouldStart(parentContext, otelRequest)) {
+          return new AdviceScope(callDepth, otelRequest, null, null);
```

Comment:
Why is the `otelRequest` here passed to the `AdviceScope`? I can't see it used anywhere. If it's not used, we shouldn't pass it to avoid confusion.

Especially because in the 5.0 instrumentation the `otelRequest` is not passed to the scope in this branch.